includes/class-wc-stripe-blocks-support.php

```diff
@@ -81,26 +93,55 @@ public function get_payment_method_script_handles() {
 	 * @return array
 	 */
 	public function get_payment_method_data() {
-		return [
-			'stripeTotalLabel'    => $this->get_total_label(),
-			'publicKey'           => $this->get_publishable_key(),
-			'allowPrepaidCard'    => $this->get_allow_prepaid_card(),
-			'title'               => $this->get_title(),
-			'button'              => [
-				'type'        => $this->get_button_type(),
-				'theme'       => $this->get_button_theme(),
-				'height'      => $this->get_button_height(),
-				'locale'      => $this->get_button_locale(),
-				'customLabel' => isset( $this->settings['payment_request_button_label'] ) ? $this->settings['payment_request_button_label'] : 'Buy now',
-			],
-			'inline_cc_form'      => $this->get_inline_cc_form(),
-			'icons'               => $this->get_icons(),
-			'showSavedCards'      => $this->get_show_saved_cards(),
-			'showSaveOption'      => $this->get_show_save_option(),
-			'supports'            => $this->get_supported_features(),
-			'stripeLocale'        => WC_Stripe_Helper::convert_wc_locale_to_stripe_locale( get_locale() ),
-			'isAdmin'             => is_admin(),
-		];
+		// We need to call array_merge_recursive so the blocks 'button' setting doesn't overwrite
+		// what's provided from the gateway or payment request configuration.
+		return array_merge_recursive(
+			$this->get_gateway_javascript_configuration(),
+			$this->get_payment_request_javascript_configuration(),
+			// Blocks-specific options
+			[
+				'title'          => $this->get_title(),
+				'icons'          => $this->get_icons(),
+				'supports'       => $this->get_supported_features(),
+				'showSavedCards' => $this->get_show_saved_cards(),
+				'showSaveOption' => $this->get_show_save_option(),
+				'isAdmin'        => is_admin(),
+				'button'         => [
+					'customLabel' => $this->payment_request_configuration->get_button_label(),
+				],
+			]
+		);
+	}
+
+	/**
+	 * Returns the Stripe Payment Gateway JavaScript configuration object.
+	 *
+	 * @return array  the JS configuration from the Stripe Payment Gateway.
+	 */
+	private function get_gateway_javascript_configuration() {
+		$js_configuration = [];
+
+		$gateways = WC()->payment_gateways->get_available_payment_gateways();
+		if ( isset( $gateways['stripe'] ) ) {
+			$js_configuration = $gateways['stripe']->javascript_configuration();
```

Comment:
Hmmm, I don't think so because this should be the `id` defined in the gateway, like this one:

https://github.com/woocommerce/woocommerce-gateway-stripe/blob/cc9e6e61335ba3bbe3a5924650612abc86f8ec43/includes/class-wc-gateway-stripe.php#L87

And trying to use that becomes sort of circular doesn't it? Instantiating a second gateway to get the ID, only to then use that ID to check for the existence of another gateway?

I agree with your sentiment though; using a string there seems fairly fragile. Maybe we should add a constant with the gateway name somewhere and use that as the ID for the gateway?

Just to be clear; `$this->name` is used to identify the payment method to the Blocks plugin I believe; it's not connected to the Payment Gateway code AFAICT 🤔

Reply:
Ah, I assumed that the `name` was already bound to be identical to the payment gateway ID for it to work, but I'm not sure if that's true – we can leave this alone 👍

Reply:
Yeah, we should use the same value for our own sanity, but it's not actually a requirement.

I'll mark this as resolved, but feel free to open the discussion on this again 👍